airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/destination_async/AsyncStreamConsumer.java

```diff
@@ -121,55 +118,32 @@ public void accept(final String messageString, final Integer sizeInBytes) throws
    * Deserializes to a {@link PartialAirbyteMessage} which can represent both a Record or a State
    * Message
    *
+   * PartialAirbyteMessage holds either:
+   * <li>entire serialized message string when message is a valid State Message
+   * <li>serialized AirbyteRecordMessage when message is a valid Record Message</li>
+   *
    * @param messageString the string to deserialize
    * @return PartialAirbyteMessage if the message is valid, empty otherwise
    */
-  private Optional<PartialAirbyteMessage> deserializeAirbyteMessage(final String messageString) {
+  @VisibleForTesting
+  public static Optional<PartialAirbyteMessage> deserializeAirbyteMessage(final String messageString) {
+    // TODO: (ryankfu) plumb in the serialized AirbyteStateMessage to match AirbyteRecordMessage code
+    // parity. https://github.com/airbytehq/airbyte/issues/27530 for additional context
```

Comment:
@edgao added the GH issue for additional context on where the deserialize/serialize to unpack the state message occurs

Reply:
https://github.com/airbytehq/airbyte/issues/27530